File: ruby/bob/bob.rb
class Bob
  def self.hey(remark)
    remark.strip!
    return 'Fine. Be that way!' if remark.empty?
    return 'Whoa, chill out!' if remark.match(/[[:alpha:]]/) && remark.upcase == remark
    return 'Sure.' if remark.chars.last == '?'
    'Whatever.'
  end
end

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    input
    |> String.trim
    |> respond
   end

  defp respond(input) do
    cond do
        String.length(input) == 0 -> "Fine. Be that way!"
        String.ends_with?(input, "?") -> "Sure."
        String.match?(input, ~r/[[:alpha:]]/) and input == String.upcase(input) -> "Whoa, chill out!"
        true -> "Whatever."
    end
  end
end
